### DISCUSSION TITLE
Why use file type over feature based architecture?

### DISCUSSION THREAD

**Discussion by crisfcodes** [infinitered/ignite#1867] [General]
I would like to know what considerations you had for choosing a [file based](https://reactjs.org/docs/faq-structure.html#grouping-by-file-type) instead [featured based](https://reactjs.org/docs/faq-structure.html#grouping-by-features-or-routes) architecture.

**Comment by jamonholmgren** [comment]
@crisfcodes Good question!

We started with strictly file-based with the original Ignite.

With [Ignite 2.0](https://www.youtube.com/watch?v=9l5_af3ryXw), we moved to a feature-based architecture. However, over the next couple years, we found that this got awkward in a few ways, and things didn't always fit "neatly" into domains or features. Of course, you have a `common` or `shared` folder, but it starts feeling a bit bloated at some point.

So with Ignite successive versions, we ended up iterating back closer to the file-based convention, but not totally. If a component, style, or other file only applies to one screen, we co-locate it there. So it's basically a hybrid approach (which they mention in that article you linked to).

Hope this helps!
